### PR TITLE
Local hail v02

### PR DESCRIPTION
**Pull request by bw2**
Updates to fix misc issues in local installs and switch over to the hail v02 pipeline. I still need to update most of the docs for how to run the new pipeline.

The basic steps for updating are:
- make sure python3 is installed
- run seqr/deploy/install_local.step8.install_pipeline_runner.sh  to install new pipeline & VEP
- double check that seqr/hail_elasticsearch_utils is on latest master (eg. cd hail_elasticsearch_utils; git pull)

To run the v02 pipeline on `my_callset.vcf.bgz`:
```
cd seqr/hail_elasticsearch_pipelines/

## step1 - annotate vcf and export to a hail matrix table file (a compressed binary format that can store variants, annotations and genotypes). VEP warnings like '...has no field...' can be ignored. 
python3 -u gcloud_dataproc/submit.py --cpu-limit 4 --num-executors 4 --hail-version 0.2  --run-locally --spark-home $SPARK_HOME  luigi_pipeline/seqr_loading.py SeqrVCFToMTTask --local-scheduler --genome-version 38 --sample-type=WGS --genome-version 38 --source-path ./my_callset.vcf.bgz --dest-path my_callset.mt'

## step2 export to an elasticsearch instance running on localhost. This will create an elasticsearch index named  'my_callset'.
python3 -u gcloud_dataproc/submit.py --cpu-limit 2 --num-executors 2 --hail-version 0.2  --run-locally --spark-home $SPARK_HOME luigi_pipeline/seqr_loading.py SeqrMTToESTask --local-scheduler --es-host localhost --es-port 9200 --es-index my_callset
```

### CODE REVIEW

**Comment by hanars** [comment]
My understanding of how dependency management works for this you should be able to just run a single command for the elsaticsearch export and it will detect that it needs to do the computations and do those. Is that correct @knguyen142 ?

**Comment by hanars** [comment]
Can you please add a manage command or at the bare minimum a helper script that sets up the correct command(s) to run? Passing a file path and string representation of a task is really unreadable/ barely usable

**Comment by knguyen142** [comment]
Yeah only `SeqrMTToESTask` is needed and it will run `SeqrVCFToMTTask` (just make sure all of `SeqrVCFToMTTask` args are included).

File: README.md [hunk 1]
```diff
@@ -33,37 +33,16 @@ For cloud-based deployments, there are Docker images and Kubernetes configs:
 **[Detailed instructions for Kubernetes deployments](deploy/KUBERNETES.md)**  
 
 
-## Updating / Migrating an older xBrowse Instance
+## Updating a local installation from the v01 to the v02 hail/elasticsearch loading pipeline:
```
Comment (hanars) [inline README.md]: I get why you added these instructions, but why did you delete the old, perfectly good instructions for upgrading other things?

File: deploy/install_local.step1.install_mongo.sh [hunk 2]
```diff
@@ -38,11 +38,15 @@ EOM
...
+    wget -qO - https://www.mongodb.org/static/pgp/server-4.2.asc | sudo apt-key add -
+
+    #sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 9DA31620334BD75D9DCB49F368818C72E52529D4
+    #echo "deb [ arch=amd64,arm64 ] https://repo.mongodb.org/apt/ubuntu xenial/mongodb-org/4.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-4.0.list
+
+    echo "deb [ arch=amd64 ] https://repo.mongodb.org/apt/ubuntu bionic/mongodb-org/4.2 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-4.2.list
     sudo apt-get update
 
-    sudo apt-get install -y mongodb-org=4.0.10 mongodb-org-server=4.0.10 mongodb-org-shell=4.0.10 mongodb-org-mongos=4.0.10 mongodb-org-tools=4.0.10
+    sudo apt-get install -y mongodb-org=4.2.1 mongodb-org-server=4.2.1 mongodb-org-shell=4.2.1 mongodb-org-mongos=4.2.1 mongodb-org-tools=4.2.1
```
Comment (hanars) [inline deploy/install_local.step1.install_mongo.sh]: why did this change?

File: deploy/install_local.step8.install_pipeline_runner.sh [hunk 3]
```diff
@@ -7,33 +7,42 @@ if [ -z "$SEQR_DIR"  ]; then
...
+    exit 1
+fi
 
-export SPARK_HOME=${SEQR_BIN_DIR}'/'${SPARK_VERSION}
 echo 'export SPARK_HOME='${SPARK_HOME} >> ~/.bashrc
+echo 'export PYSPARK_PYTHON=python3' >> ~/.bashrc
+echo 'export LUIGI_CONFIG_PATH=luigi_pipeline/configs/seqr-loading-local-GRCh38.cfg'  >> ~/.bashrc
+
+# set up gcloud connector
+sudo python3 -m pip install git+https://github.com/bw2/hail-utils.git
```
Comment (hanars) [inline deploy/install_local.step8.install_pipeline_runner.sh]: Why is this not in a maclab repo?

File: deploy/install_local.step8.install_pipeline_runner.sh [hunk 4]
```diff
@@ -47,67 +56,96 @@ sudo $(which pip) install --ignore-installed decorator==4.2.1
...
-
-mkdir -p ${SEQR_DIR}/vep/loftee_data_grch37 ${SEQR_DIR}/vep/loftee_data_grch38 ${SEQR_DIR}/vep/homo_sapiens
-sudo ln -s ${SEQR_DIR}/vep /vep
-sudo chmod -R 777 /vep
-
-if [ ! -f /usr/local/bin/perl ]
-then
-    sudo ln -s /usr/bin/perl /usr/local/bin/perl
-fi
+# Download and install VEP GRCh38 - steps based on gs://hail-common/vep/vep/vep95-loftee-1.0-GRCh38-init-docker.sh
```
Comment (hanars) [inline deploy/install_local.step8.install_pipeline_runner.sh]: I don't think this comment explains whats happening here anymore?

**Comment by hanars** [comment]
I feel like to update a submodule to a more recent version of the repo you have to make a code change? Am I wrong about that, or is there a change we need to make?

File: README.md [hunk 5]
```diff
@@ -3,47 +3,22 @@ seqr
...
+The seqr production instance runs on Google Kubernetes Engine (GKE) and data is loaded using Google Dataproc Spark clusters. 
   
-**[Detailed instructions for local installations](deploy/LOCAL_INSTALL.md)**  
-
-
-## Updating / Migrating an older seqr Instance
-
-For notes on how to update an older instance, see  
-
-[Update/Migration Instructions](deploy/MIGRATE.md)
```
Comment (hanars) [inline README.md:36]: Why take out this section?

File: deploy/MIGRATE.md [hunk 6]
```diff
@@ -1,28 +0,0 @@
-This README describes steps for migrating an older seqr instance.
```
Comment (hanars) [inline deploy/MIGRATE.md:1]: Why delete this?

File: deploy/deploy-requirements.txt [hunk 7]
```diff
@@ -1,6 +0,0 @@
-pip                              # make sure the latest version of pip is installed
-psutil==5.6.7                    # library for process and system monitoring
-
-# kubernetes deployment dependencies
-Jinja2==2.10.1                   # jinja templates are used in the kubernetes deployment scripts
```
Comment (hanars) [inline deploy/deploy-requirements.txt:5]: Don't we still need these for servctl?

File: deploy/docker/phenotips/entrypoint.sh [hunk 8]
```diff
@@ -9,8 +9,16 @@ echo Current Directory: $(pwd)
...
-# init PhenoTips db
-PGPASSWORD=xwiki psql --host postgres --port 5432 -U xwiki xwiki -f /init_phenotips_db.sql
+# init phenotips db unless it already exists
+if ! psql --host postgres -U postgres -l | grep xwiki; then
+
+  psql --host postgres -U postgres -c 'CREATE DATABASE xwiki';
+  psql --host postgres -U postgres -c "CREATE ROLE xwiki WITH CREATEDB LOGIN PASSWORD 'xwiki'";
+  psql --host postgres -U postgres -c 'GRANT ALL PRIVILEGES ON DATABASE xwiki TO xwiki';
+
+  PGPASSWORD=xwiki psql --host postgres -U xwiki xwiki -f /init_phenotips_db.sql
```
Comment (hanars) [inline deploy/docker/phenotips/entrypoint.sh:19]: Shouldn't this always get run even if the DB already exists?

File: deploy/docker/seqr/entrypoint.sh [hunk 9]
```diff
@@ -35,6 +35,16 @@ if [ $SEQR_GIT_BRANCH ]; then
   git checkout $SEQR_GIT_BRANCH
 fi
 
+# init seqrdb unless it already exists
+if ! psql --host postgres -U postgres -l | grep seqrdb; then
+
+  psql --host postgres -U postgres -c 'CREATE DATABASE seqrdb';
+  psql --host postgres -U postgres -c 'CREATE DATABASE reference_data_db';
+  psql --host postgres -U postgres reference_data_db <  <(curl -s https://storage.googleapis.com/seqr-reference-data/gene_reference_data_backup.gz | gunzip -c -);
```
Comment (hanars) [inline deploy/docker/seqr/entrypoint.sh:43]: I would strongly prefer if we use a variable to reference the backup instead of a hardcoded string here

File: docker-compose.yml [hunk 10]
```diff
@@ -0,0 +1,107 @@
+version: '2.4'
+
+services:
+  postgres:
+    image: gcr.io/seqr-project/postgres:gcloud-dev
```
Comment (hanars) [inline docker-compose.yml:5]: We should build the new images and change the references before deploying

File: README.md (hunk 5 above)
Comment (bw2) [reply README.md:36]: sorry, thought this was out date.

File: deploy/docker/phenotips/entrypoint.sh (hunk 8 above)
Comment (bw2) [reply deploy/docker/phenotips/entrypoint.sh:19]: That script creates various tables. It only needs to run once.

File: deploy/docker/seqr/entrypoint.sh (hunk 9 above)
Comment (bw2) [reply deploy/docker/seqr/entrypoint.sh:43]: This path will remain the same for all installs. Are you suggesting a variable defined at the top of this script?
Also, at some point this file should have a version.

File: docker-compose.yml (hunk 10 above)
Comment (bw2) [reply docker-compose.yml]: done

File: deploy/docker/seqr/entrypoint.sh (hunk 9 above)
Comment (hanars) [reply deploy/docker/seqr/entrypoint.sh:43]: Yeah I just meant a variable at the top of the script

Comment (hanars) [reply deploy/docker/seqr/entrypoint.sh:43]: I mean ideally I'd rather it be a kubernetes variable and we could just set it in the shared YAML, but at a minimum a variable in the script would be nice

File: docker-compose.yml [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+services:
+  postgres:
+    image: gcr.io/seqr-project/postgres:gcloud-prod
+    ports:
+      - 5432:5432
+    volumes:
+      - ./data/postgres:/var/lib/postgresql/data
+    healthcheck:
+      test: pg_isready -h postgres -U postgres; #bash -c "psql -h localhost -U xwiki -d xwiki -c 'BEGIN; END;'";
```
Comment (hanars) [inline docker-compose.yml:11]: maybe take out the commented code?

File: requirements.txt [hunk 12]
```diff
@@ -7,6 +7,7 @@ django-cors-headers==3.0.2       # allows CORS requests for client-side developm
 elasticsearch==6.4.0             # elasticsearch client
 elasticsearch-dsl==6.4.0	     # elasticsearch query utilities
 gunicorn==19.10.0                # web server
+Jinja2==2.10.1                   # jinja templates are used in the kubernetes deployment scripts
```
Comment (hanars) [inline requirements.txt:10]: I'd rather this go in requirements-dev and keep this file for runtime dependencies in seqr, but if you feel strongly we can leave it here
